graphs/a_star.py

```diff
@@ -16,6 +16,30 @@ def search(
     cost: int,
     heuristic: list[list[int]],
 ) -> tuple[list[list[int]], list[list[int]]]:
+    """
+    Search for a path on a grid avoiding obstacles.
+    >>> grid = [[0, 1, 0, 0, 0, 0], \
+                [0, 1, 0, 0, 0, 0], \
+                [0, 1, 0, 0, 0, 0], \
+                [0, 1, 0, 0, 1, 0], \
+                [0, 0, 0, 0, 1, 0]]
+    >>> init = [0, 0]
+    >>> goal = [len(grid)-1, len(grid[0])-1]
+    >>> cost = 1
+    >>> heuristic = [[0 for row in range(len(grid[0]))] for col in range(len(grid))]
+    >>> for i in range(len(grid)):
+    ...     for j in range(len(grid[0])):
+    ...         heuristic[i][j] = abs(i - goal[0]) + abs(j - goal[1])
+    ...         if grid[i][j] == 1:
+    ...             heuristic[i][j] = 99
+    >>> path, action = search(grid, init, goal, cost, heuristic)
+    >>> print(path)
+    [[0, 0], [1, 0], [2, 0], [3, 0], [4, 0], [4, 1], [4, 2],\
+ [4, 3], [3, 3], [2, 3], [2, 4], [2, 5], [3, 5], [4, 5]]
```

Comment:
```suggestion
    >>> path
    [[0, 0], [1, 0], [2, 0], [3, 0], [4, 0], [4, 1], [4, 2],
    ... [4, 3], [3, 3], [2, 3], [2, 4], [2, 5], [3, 5], [4, 5]]
```

Reply:
Check this in the CI: https://github.com/TheAlgorithms/Python/actions/runs/6517757487/job/17702529065?pr=10397

Reply:
Nice @tianyizheng02 Thanks!